src/coreComponents/linearAlgebra/interfaces/hypre/HypreMatrix.cpp

```diff
@@ -354,18 +436,34 @@ void HypreMatrix::set( globalIndex const rowIndex,
                                                   values ) );
 }
 
-void HypreMatrix::insert( globalIndex const rowIndex,
+void HypreMatrix::insert( globalIndex const rowIndex0,
                           globalIndex const * colIndices,
                           real64 const * values,
                           localIndex size )
 {
   GEOSX_LAI_ASSERT( insertable() );
 
-  HYPRE_Int ncols = LvArray::integerConversion< HYPRE_Int >( size );
+#if defined(GEOSX_USE_CUDA) && defined(OVERRIDE_CREATE)
+  array1d< globalIndex > rowIndexDevice( 1 );
+  array1d< HYPRE_Int > ncolsDevice( 1 );
+
+  rowIndexDevice[0] = rowIndex0;
+  ncolsDevice[0] = LvArray::integerConversion< HYPRE_Int >( size );
+
+  rowIndexDevice.move( LvArray::MemorySpace::GPU, false );
+  ncolsDevice.move( LvArray::MemorySpace::GPU, false );
+
+  globalIndex const * const rowIndex = rowIndexDevice.data();
+  HYPRE_Int * const ncols = ncolsDevice.data();
+#else
+  globalIndex const * const rowIndex = &rowIndex0;
+  HYPRE_Int * const ncols = reinterpret_cast< HYPRE_Int * >(&size);
```

Comment:
This is only called if we are using hypre with cuda...which means that they are the same.

Reply:
Do you need to cast then? IDK if you have a check somewhere else to make sure they're the same type.

Reply:
These checks don't require the same type, but only that their size and signed-ness match. So you could have e.g. a 32-bit `int` and a 32-bit `long` that match in binary for all purposes, but still treated as different types and you have to `reinterpret_cast` the pointer. We could also make the check stricter and require the same type, don't know if it would break some configurations?

Reply:
I think we do actually why hard type checks at some point. Avoiding `reinterpret_cast` is generally a good policy.